apps/api/src/app/gene/gene.service.ts

```diff
@@ -0,0 +1,31 @@
+import {Injectable, OnModuleInit} from '@nestjs/common';
+import {PrismaService} from "../prisma.service";
+
+@Injectable()
+export class GeneService {
+  constructor(private prismaService: PrismaService) {}
+
+  async fetchGeneSymbols() {
+    try {
+      const hugoJSON = await this.prismaService.hugo.findMany();
+      const symbols = hugoJSON.map((entry) => entry.gene);
+      return symbols;
+    } catch (error) {
+      console.error('Error fetching', error);
+      throw error;
+    }
+  }
+
+  async filterGeneSymbols(query: string) {
+    try {
+      const symbols = await this.fetchGeneSymbols();
+      const filteredSymbols = symbols.filter((symbol) =>
+        symbol.toLowerCase().startsWith(query?.toLowerCase())
```

Comment:
If query could be undefined, let's just add a check at the beginning of the method and return empty list to avoid unnecessary DB query. The condition should also include empty string.